services/collaboration/src/controllers/roomController.ts

```diff
@@ -0,0 +1,186 @@
+import { Request, Response } from 'express';
+import {
+    createRoomInDB,
+    createYjsDocument,
+    deleteYjsDocument,
+    findRoomById,
+    findRoomsByUserId,
+    closeRoomById,
+    updateRoomUserStatus,
+} from '../services/mongodbService';
+import axios from 'axios';
+import { handleNotFound, handleSuccess, handleServerError, handleBadRequest } from '../utils/helper';
+import config from '../config';
+
+/**
+ * Create a room with users, question details, and Yjs document
+ * @param user1
+ * @param user2
+ * @param topics
+ * @param difficulty
+ * @returns roomId
+ */
+export const createRoomWithQuestion = async (user1: any, user2: any, topics: string[], difficulty: string) => {
```

Comment:
I think this is quite liberal usage of `any`, ideally we should have interfaces for objects like user so that code is more readable and typescript can spot any silly mistakes